### DISCUSSION TITLE
Mindstorm inventor Ultrasonic distance sensor sound cone configuration

### DISCUSSION THREAD

**Discussion by mohankrr** [pybricks/support#564] [Q&A] [answered]
The distance sensor is not reliable unless it’s facing a flat surface. Is there a way to narrow the sound cone using config parameters to make it a bit more reliable for specific usecases?

**Comment by dlech** [comment, marked as answer]
> The distance sensor is not reliable unless it’s facing a flat surface.

This is because of the laws of physics. Even industrial ultrasonic sensors have this problem. If the sound waves are not reflected directly back to the sensor, then the signal is too weak and a measurement cannot be made.

> Is there a way to narrow the sound cone using config parameters to make it a bit more reliable for specific usecases?

The only configuration is a "fast" sensing mode ([specs](https://education.lego.com/v3/assets/blt293eea581807678a/blt64c2b9534cf10f68/5f8801b8bc43790f5c4389ea/techspecs_technicdistancesensor.pdf)), but this only changes the time that the sensor waits to hear the echo. In our tests, we didn't get good results with this mode, so we did not use it in Pybricks (although it should be possible to use this mode using the [PUPDevice](https://docs.pybricks.com/en/stable/iodevices/pupdevice.html) class).

**Reply by mohankrr** [reply]
Thanks.
